### DISCUSSION TITLE
Carddav error due to Bday

### DISCUSSION THREAD

**Discussion by testmodem** [YetiForceCompany/YetiForceCRM#14364] [General]
Dear all
Everything is working but i am getting only one error when i try to sync my contacts from Davx5 due to Bday field. The Bday in contacts is specified as **1983-02-22**, However Yetiforce need is in **22-02-1983** format i guess. Below is a sample contact from my mobile contact list.

_FN:Test1
TEL;CELL:00441111111111
TEL;CELL:00441111111111
EMAIL;X-INTERNET:abc@abc.com
BDAY:1983-02-22
END:VCARD
BEGIN:VCARD
VERSION:2.1_

When I remove 1983-02-22 from Bday field, I can import the contact successfully.

Thanks

**Comment by mariuszkrzaczkowski** [comment]
![github](https://user-images.githubusercontent.com/10198654/110434137-b76ef100-80b1-11eb-8970-c7dcc8102aee.jpg)

**Comment by mariuszkrzaczkowski** [comment]
show this error

**Reply by testmodem** [reply]
Thank you dear mariuszkrzaczkowski .. here are the further information.

1) Server version: Apache/2.4.46 (Ubuntu), Server built:   2020-08-25T12:13:38
2) importing contacts to DAVx5 is successfull without any problem, error occurs when syncing it with yetiforce.

3) This is the screenshot of error showing at site.com/cron.php
![Carddav DOB error](https://user-images.githubusercontent.com/59494950/110455019-bfe31e00-80e9-11eb-9701-22ea9ed604c7.png)

4) http://site.com/dav.php/addressbooks/admin1/YFAddressBook/  shows 5071 contacts syncronized.
![yfaddressbook](https://user-images.githubusercontent.com/59494950/110456643-a0e58b80-80eb-11eb-8d5d-c892a639476c.png)

5) http://site.com/index.php?module=Contacts   hangs at 3042 
![contacts module](https://user-images.githubusercontent.com/59494950/110457152-26693b80-80ec-11eb-899c-5229ddbf0718.png)



Thanks

**Reply by mariuszkrzaczkowski** [reply]
Include all required information
![github](https://user-images.githubusercontent.com/10198654/110457594-168d2f80-80cb-11eb-8cea-97b55c4849af.jpg)

**Reply by testmodem** [reply]
Thank you dear mariuszkrzaczkowski .. here are the further information.

1) Server version: Apache/2.4.46 (Ubuntu), Server built:   2020-08-25T12:13:38, using chrome .
2) importing contacts to DAVx5 is successfull without any problem, error occurs when syncing it with yetiforce.

3) This is the screenshot of error showing at site.com/cron.php
![Carddav DOB error](https://user-images.githubusercontent.com/59494950/110455019-bfe31e00-80e9-11eb-9701-22ea9ed604c7.png)

4) http://site.com/dav.php/addressbooks/admin1/YFAddressBook/  shows 5071 contacts syncronized.
![yfaddressbook](https://user-images.githubusercontent.com/59494950/110456643-a0e58b80-80eb-11eb-8d5d-c892a639476c.png)

5) http://site.com/index.php?module=Contacts   hangs at 3042 
![contacts module](https://user-images.githubusercontent.com/59494950/110457152-26693b80-80ec-11eb-899c-5229ddbf0718.png)

6) Configuration settings.
![config settings](https://user-images.githubusercontent.com/59494950/110460200-b3fa5a80-80ef-11eb-91ab-d2614378bae1.png)

7) Full system config
![screencapture-grandhouse-index-php-2021-03-09-15_59_32](https://user-images.githubusercontent.com/59494950/110460914-9f6a9200-80f0-11eb-9076-cbfd58c08dbb.png)

**8)** Error.log .. not sure if its relevant.
E_WARNING: file_get_contents(/var/www/yetiforce/cache/session/sess_0qgb1amemi99tlmqop1gu4131a): failed to open stream: Permission denied in /var/www/yetiforce/app/Session/File.php, line 23
#0 app/Session/File.php (23)  >>  file_get_contents('cache/session/sess_0qgb1amemi99tlmqop1gu4131a')
#1 modules/Vtiger/crons/SessionCleaner.php (28)  >>  App\Session\File->clean()
#2 cron.php (98)  >>  Vtiger_SessionCleaner_Cron->process()
E_WARNING: file_get_contents(/var/www/yetiforce/cache/session/sess_kpcli7orng6r77un4s8071onra): failed to open stream: Permission denied in /var/www/yetiforce/app/Session/File.php, line 23
#0 app/Session/File.php (23)  >>  file_get_contents('cache/session/sess_kpcli7orng6r77un4s8071onra')
#1 modules/Vtiger/crons/SessionCleaner.php (28)  >>  App\Session\File->clean()
#2 cron.php (98)  >>  Vtiger_SessionCleaner_Cron->process()

9)Nothing seem relevant in system.log





Thanks

**Reply by mariuszkrzaczkowski** [reply]
What is your date format of the user

**Reply by testmodem** [reply]
Thanks change the user date format and its working .. Thanks
http://site.com/index.php?module=Users&parent=Settings&view=Detail&record=1

**Reply by vovpff** [reply]
@testmodem

Could you share for me your api.php and other dav configuration data? I'm also cant establish sync between dawdroid an YF

**Reply by testmodem** [reply]
```php
namespace Config;

/**
 * Configuration Class.
 */
class Api
{
	/** List of active services. Available: dav, webservice */
	public static $enabledServices = ['dav'];

	/** Dav configuration. Available: false, true */
	public static $enableBrowser = true;

	/** Dav configuration. Available: false, true */
	public static $enableCardDAV = true;

	/** Dav configuration. Available: false, true */
	public static $enableCalDAV = true;

	/** Dav configuration. Available: false, true */
	public static $enableWebDAV = false;

	/** Webservice config. Enabling contact notifications about the new account in the portal. Available: false, true */
	public static $enableEmailPortal = true;

	/** Webservice config. Available: false, true */
	public static $ENCRYPT_DATA_TRANSFER = false;

	/** Webservice config. */
	public static $AUTH_METHOD = 'Basic';

	/** Webservice config. */
	public static $PRIVATE_KEY = 'config/private.key';

	/** Webservice config. */
	public static $PUBLIC_KEY = 'config/public.key';
}
```

**Reply by mariuszkrzaczkowski** [reply]
@vovpff  see https://www.youtube.com/watch?v=Jg3mDVPz5dg&list=PLX0lktP3Kukg_Da-TaEBys-aON68Yy2vh&index=16

**Reply by mariuszkrzaczkowski** [reply]
> Thanks change the user date format and its working .. Thanks
> http://site.com/index.php?module=Users&parent=Settings&view=Detail&record=1

fixed https://github.com/YetiForceCompany/YetiForceCRM/commit/daff1fd9d230c7a5178eaef5e53a6c5d47b82772

**Reply by testmodem** [reply]
Can you please help me further. after successful import I am facing two problems
**P1) some of my contacts are rename for example.**
**Original Contacts Names and Number**
Yeti Demo, Number 111
Yeti Demo 2, Number 112
Yeti Demo 3, Number 113
Yeti Demo 4, Number 114

**Rename To**
Yeti Demo, Number 111
Yeti 2, Number 112
Yeti 3, Number 113
Yeti 4, Number 114

**P2) What file can i configure to increase maximum character limit for Contact name ?**

Thanks

**Reply by mariuszkrzaczkowski** [reply]
you would have to change in vtiger field and increase table column
![image](https://user-images.githubusercontent.com/10198654/135853488-84116e4a-abf4-44ac-85fb-a29d3bc4c16b.png)
